src/message.rs

```diff
@@ -289,32 +321,14 @@ WHERE id=?;
         if !msg.rfc724_mid.is_empty() {
             ret += &format!("\nMessage-ID: {}", msg.rfc724_mid);
 
-            let server_uids = context
-                .sql
-                .query_map(
-                    "SELECT folder, uid FROM imap WHERE rfc724_mid=?",
-                    (msg.rfc724_mid,),
-                    |row| {
-                        let folder: String = row.get("folder")?;
-                        let uid: u32 = row.get("uid")?;
-                        Ok((folder, uid))
-                    },
-                    |rows| {
-                        rows.collect::<std::result::Result<Vec<_>, _>>()
-                            .map_err(Into::into)
-                    },
-                )
-                .await?;
+            let server_uids = Self::get_info_server_uids(context, msg.rfc724_mid).await?;
 
             for (folder, uid) in server_uids {
                 // Format as RFC 5092 relative IMAP URL.
                 ret += &format!("\n</{folder}/;UID={uid}>");
             }
         }
-        let hop_info: Option<String> = context
-            .sql
-            .query_get_value("SELECT hop_info FROM msgs WHERE id=?;", (self,))
-            .await?;
+        let hop_info: Option<String> = self.hop_info(context).await?;
```

Comment:
`: Option<String>` isn't needed here. And i still suggest to remove Option at all, it's just a human-readable info